src/library/trackset/crate/cratefeature.cpp

```diff
@@ -517,30 +536,50 @@ QModelIndex CrateFeature::rebuildChildModel(CrateId selectedCrateId) {
     m_pSidebarModel->removeRows(0, pRootItem->childRows());
 
     QList<TreeItem*> modelRows;
-    modelRows.reserve(m_pTrackCollection->crates().countCrates());
+    modelRows.reserve(m_pTrackCollection->crates().countCrates(true) + 1);
+
+    auto archiveItem = new TreeItem(tr("Archived"), -1);
+    bool selectedIsArchived = false;
 
     int selectedRow = -1;
     CrateSummarySelectResult crateSummaries(
             m_pTrackCollection->crates().selectCrateSummaries());
     CrateSummary crateSummary;
     while (crateSummaries.populateNext(&crateSummary)) {
         auto pTreeItem = newTreeItemForCrateSummary(crateSummary);
-        modelRows.append(pTreeItem.get());
-        pTreeItem.release();
+        if (crateSummary.isArchived()) {
+            archiveItem->appendChild(std::move(pTreeItem));
+        } else {
+            modelRows.append(pTreeItem.get());
+            pTreeItem.release();
+        }
         if (selectedCrateId == crateSummary.getId()) {
-            // save index for selection
-            selectedRow = modelRows.size() - 1;
+            if (crateSummary.isArchived()) {
+                selectedIsArchived = true;
+                selectedRow = archiveItem->childRows() - 1;
+            } else {
+                selectedRow = modelRows.size() - 1;
+            }
         }
     }
-
+    modelRows.append(archiveItem);
+    // Save the size of the model now because it gets cleared when the items are inserted
+    // into the model.
+    const int rowCount = modelRows.size();
     // Append all the newly created TreeItems in a dynamic way to the childmodel
     m_pSidebarModel->insertTreeItemRows(modelRows, 0);
 
     // Update rendering of crates depending on the currently selected track
     slotTrackSelected(m_selectedTrackId);
 
     if (selectedRow >= 0) {
-        return m_pSidebarModel->index(selectedRow, 0);
+        if (selectedIsArchived) {
+            QModelIndex archiveIndex = m_pSidebarModel->index(rowCount - 1, 0);
+            m_pSidebarWidget->setExpanded(archiveIndex, true);
+            return m_pSidebarModel->index(selectedRow, 0, archiveIndex);
```

Comment:
In my use, I find it not very convenient to keep the archived crate selected.  Instead I think it might be nicer to select the crate below where it used to be in the hierarchy.  This is because archiving a crate is kind of like "please make this go away" so having the selection follow the archive process doesn't quite make sense.

Reply:
But why expand the Archive item?